Feature/restaurants module

talorlangnas-moveo opened the pull request:
Implemented Restaurants module with functionality to create a restaurant, fetch all restaurants, and fetch a restaurant by ID

talorlangnas-moveo committed:
working on RestaurantsModule - in process

talorlangnas-moveo committed:
creating fetch all function

- Updated RestaurantsController to include a space in the constructor for consistency.
- Enhanced RestaurantsModule to import MongooseModule and define the Restaurant schema.
- Refactored RestaurantsService to inject the Restaurant model and implement the create method for adding new restaurants to the database.

talorlangnas-moveo committed:
Adding create, fetch all and fetch by id functions

- Updated RestaurantsController to use async/await in findAll method and renamed findOne to findById for clarity.
- Enhanced RestaurantsService with async findAll and findById methods, including error handling for not found scenarios.

talorlangnas-moveo committed:
Refactor InfoPanel component and styles

- Updated InfoPanel to support conditional rendering of buttons based on desktop view.
- Added new prop `displayButtonDesktop` to control button visibility on desktop.
- Adjusted styles in infoPanel.module.scss for better layout and spacing.
- Introduced a new link style for button links.

talorlangnas-moveo committed:
refactor: move App and Restaurant modules to modules directory

talorlangnas-moveo committed:
refactor: remove update and delete methods from RestaurantsController and RestaurantsService

- Removed unused update and delete methods from both RestaurantsController and RestaurantsService for cleaner code.
- Simplified constructor formatting in RestaurantsController.

talorlangnas-moveo committed:
refactor: enhance error handling in findById method of RestaurantsService

- Added validation for restaurant ID format using BadRequestException.
- Simplified error handling logic for not found scenarios in findById method.

talorlangnas-moveo committed:
Merge branch 'development' into feature/restaurants_module

taluliel18 commented on epicure-server/src/modules/restaurants/dto/create-restaurant.dto.ts:
```diff
@@ -0,0 +1,39 @@
+
+  @IsString()
+  @IsOptional()
+  homePageImgUrlDesktop?: string;
```
Why do we need three image URLs?

talorlangnas-moveo replied:
done.
Deleted  homePageImgUrlMobile and homePageImgUrlDesktop.

taluliel18 commented on epicure-server/src/modules/restaurants/dto/create-restaurant.dto.ts:
```diff
@@ -0,0 +1,39 @@
+export class CreateRestaurantDto {
+  @IsString()
+  @IsNotEmpty()
+  title: string;
```
It’s clearer to use 'name' for this attribute instead of 'title'

talorlangnas-moveo replied:
done

talorlangnas-moveo committed:
refactor: remove unused homepage images and update restaurant data structure

- Deleted mobile and desktop homepage images for the Claro restaurant.
- Updated the restaurant data structure by removing references to the deleted images.
- Changed the restaurant title property to name in the CreateRestaurantDto and Restaurant schema for consistency.
- Adjusted the DishesDisplay component to use a single image URL instead of separate desktop and mobile URLs.
